utils/matrix/setup_matrix.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python
+from __future__ import print_function
+
+import argparse
+import os
+import shutil
+
+try:
+    from configparser import ConfigParser
+except ImportError:
+    from six.moves import configparser
+    import six
+
+    if six.PY2:
+        ConfigParser = configparser.SafeConfigParser
+    else:
+        ConfigParser = configparser.ConfigParser
+
+from shared import get_logger, check_call
+
+
+# build targets from
+# https://mpas-dev.github.io/compass/latest/developers_guide/machines/index.html#supported-machines
+all_build_targets = {
+    'anvil': {
+        ('intel', 'impi'): 'intel-mpi',
+        ('intel', 'openmpi'): 'ifort',
+        ('intel', 'mvapich'): 'ifort',
+        ('gnu', 'openmpi'): 'gfortran',
+        ('gnu', 'mvapich'): 'gfortran'},
```

Comment:
Is there some way we could choose a subset of these in the config file?

Reply:
You don't need to choose in the config file.  You choose by how you call `./conda/configure_compass_env.py`.  You can do:
```
./conda/configure_compass_env.py --compiler all --mpi all ...
```
to get all of these.  Or you can do:
```
./conda/configure_compass_env.py --compiler intel --mpi all ...
```
to get just the intel ones or:
```
./conda/configure_compass_env.py --compiler all --mpi openmpi ...
```
to get just the openmpi ones or:
```
./conda/configure_compass_env.py --compiler intel ...
```
(omitting the `--mpi`) to get the default MPI library for intel or:
```
./conda/configure_compass_env.py --compiler intel intel intel gnu --mpi impi openmpi mvapich mvapich ...
```
to get a set with 3 intel variants and a gnu variant.

Do you want to make a suggestion for how to add this to the README here?  Or the docs somewhere else?  I wasn't sure how to handle it because it's a feature I expect most developers not necessarily to need (so I didn't want to overwhelm the quick start for developers with this) but it isn't directly related to the matrix utility in that it should work fine even if you build, setup and run manually.

Reply:
I will add something similar to my comment above to the README for now.  It can make its way to the main docs later...

Reply:
This is definitely helpful. My thought was that just because you have these environments created, doesn't mean you want to necessarily include them in the matrix. Can you edit the `matrix.log` file to exclude certain ones?

Reply:
You could do that but maybe just call the config script again? It won't delete anything, just make sure everything is up to date and write the matrix.log you want.

Reply:
I will think about this some more...

Reply:
That's a good idea.